tempesta_fw/connection.c

```diff
@@ -94,6 +94,8 @@ tfw_connection_release(TfwConnection *conn)
 int
 tfw_connection_send(TfwConnection *conn, TfwMsg *msg)
 {
+	if (conn->tls)
+		return tfw_tls_send_msg(conn->tls, msg);
 	return ss_send(conn->sk, &msg->skb_list, msg->ss_flags);
```

Comment:
Ok, this has sense: we can think about `TfwConnection` as 6th (presentation) OSI layer on top of TCP sockets, so the layer can be encrypted or not. I think we don't need to maintain two TfwConnection-like interfaces for encrypted and raw data.

Please update `connection.h` header comment: replace `...generic connection (at OSI level 4) management` by `...generic connection management at OSI level 6 (presentation)`